### ISSUE TITLE
date-picker: number of watchers over 300!

### ISSUE THREAD

**Issue by demetriusnunes** [lumapps/lumX#211]
Guys,

Recently I've been optimizing our app that uses lumx. On a certain page with a simple form we noticed it already started with over 300 watchers. After digging a little, I found out that most of them was coming from lumX date-picker component, even without it showing. This only for one single date-picker.

I believe this can be heavily optimized, specially if the date picker is hidden.

I did a simple test of adding a ng-if to the part of the template related to the dialog and it's already a big improvement.

I'm using the awesome tool ng-stats for inspecting that:
https://github.com/kentcdodds/ng-stats

**Comment by yconoclast** [comment]
Looking at the demo page, it has 904 watchers

![lumx](https://cloud.githubusercontent.com/assets/4692941/6837804/d539592c-d348-11e4-924f-6d8f4ddcc9f7.png)

**Comment by demetriusnunes** [comment]
@yconoclast which tool is this that you're using?

**Comment by yconoclast** [comment]
Angular Watchers
https://chrome.google.com/webstore/detail/angular-watchers/nlmjblobloedpmkmmckeehnbfalnjnjk

**Comment by leppie** [comment]
This is mainly due to years being currentyear +- 100 and the year selection view. For my use, I dont need that.

**Comment by malexandre** [comment]
The watchers count is too damn high, and I'll look into it really soon! Using ng-show is a first step, but we need to optimize the overall performance too.

**Comment by malexandre** [comment]
I finally found time to work on this. I made enhancements to the date-picker (and a bunch of other elements). What's important here, is in the date-picker : Better scope & ng-if management, which make the numbers of watchers really go down, and avoid memory leaks.

With this, I think everyone should see a great bump in the performance, and there is way less watchers. Having multiple date picker on the same page is not a hassle anymore.

Thank you for the report!
